pkg/internal/controller/controller.go

```diff
@@ -210,8 +210,12 @@ func (c *Controller) processNextWorkItem() bool {
 
 		return false
 	} else if result.Requeue {
-		c.Queue.AddRateLimited(req)
-		return true
+		if result.RequeueAfter != nil {
+			c.Queue.AddAfter(req, *result.RequeueAfter)
```

Comment:
Should we be forgetting with `AddAfter`?  This currently does...

Reply:
Good catch.  Done